### ISSUE TITLE
Chapter 4 - Following through the course, I do not see where it was taught to be able to accomplish exercise 4.6....

### ISSUE THREAD

**Issue by 4Python3** [csev/py4e#283]
As stated in the title...Chapter 4 - Following through the course, I do not see where it was taught to be able to accomplish exercise 4.6.  I mean, Chapter 4 is very confusing, not thought out very well from a, "teaching to new student", aspect.  Dr. Chuck goes confusingly and quickly goes through basically how to use "input", more so than concentrating on building and understanding a function...missing at least what is needed to create code for solving chapter 4.6's exercise.  I have watched the videos multiple times trying to see what I missed, however, after reviewing several times, I still do not see where he teaches how to construct a program to accomplish 4.6.  I go to the assignment and all of a sudden I see the following with NO further instruction on whether some of this is correct, it is all correct by out of order, it is just an example, etc.:

 def computepay(h,r):
    return 42.37

hrs = input("Enter Hours:")
p = computepay(10,20)
print("Pay",p)

Up to this point in the course, I do not see where he talked about this.  I see line 1...def computepay(h, r).  I know that this is saying create a function called computepay and run it when called and use the the arguments "h 8 r" that is going to be received from the user's input.  Line 2,..why is there this 42.36 in there?  I mean, how am I supposed to know what to do with that?  That was never talked about.  Am I supposed to change those numbers?  Should the function contain those numbers.  Seems as though they are there and will always return that same value....that doesn't make any sense and there is no way to ask Dr. Chuck.  Line 3 I get....ask the user for the hours.  Ok, but again...is this incomplete code?  Is what is given when resetting the code the first few lines of code that we should use unchanged, or do we need to add code in between the line of what is given.  Meaning....is prompts the user for the hours, but I do not see where it prompts for the rate...so guessing that we are supposed to alter even what was given and add code in between the lines.  I dunno.

Line 4...p = computepay(10,20).  Again, I am confused, if the def statement says to take, "h & r", from the user input, why are there permanent numbers already in the argument area here?  What are these number?  What is 10 & 20?  

I have been following along in depth and have 100% everything up to this point, but none of this chapter 4 assignment makes any sense, for the teaching on this chapter.  It was very vague, very messy, and very confusing. 

At the beginning of (I believe) this chapter, he stated that there were 4 types...sequential, conditional, something else, and now this, however, throughout the study, he has managed to skip talking about while statements, etc., but at the start of the chapter, it appears that he thinks that he has gone over all 4 (unaware that he skipped while statements, etc).  I fear that Dr. Chuck is losing track that he is teaching people with NO programming experience and gets messy.  I feel that he talk about things that are not in the correct context of what we are studying.  Meaning for example...teaching about functions by the chapter seems more that it is teaching to use input in function building.  I am lost and another point, It took me forever to find a place like this where I could HOPEFULLY get some help and I am still not even sure if I am in the right area for getting help.  Meaning, I could not find an acceptable way of asking for help.  I know there is a forum somewhere.  I know that there is a place where I can ask someone to come in and look at my code, but for the life of me, I cannot not find how to do it.  It should not be that hard.  There should be a quick and easy to find link somewhere, especially in the assignment page, but I do not see it.  Ridiculous.  Then the autograder assignment area....you run the code (which mine seems to always give the correct answer), but somehow it is wrong.  I get, "error storing grade on server"....that is it.  ???  It doesn't tell me why and just let to stare at a page that I do not know how to fix.  I mean...to me...my code looks right (but obviously is not), and it seems to work and give the correct answer, but autograder says something is not right, and I do not know why, nor can I figure out what I am doing wrong even after going back numerous times and reviewing the teaching.  So what am I supposed to do, just quit?  Where am I supposed to get the answer or a better understanding on how to do this if it is not in the teaching video?  More so, why is it not fully explained in the teaching video if you are going to test me on it or give me an assignment that has to be passed before moving on?  Ridiculous!!!

This is so confusing.  I don't even know if filling this out in GitHub has anything to do with my Python course or for getting help with my course.  This is crazy and a waste of time.

**Comment by IzuTJ** [comment]
Most of the pre-written code in auto-grader exercises is just there as a 'boiler-plate' meant to guide you and provide a rough structure that you might use. You can delete the whole thing and write it all by yourself if you wish to do that.

What you need to solve this exercise is arithmetic, operator precedence (covered in chapter 2 - variables) and if else statements (covered in the chapter you are mentioning).
And for chapter 5 we don't need while loops to solve any problem. They will make their appearance in the next chapter.

I hope I was able to help a little.

**Comment by 4Python3** [comment]
Well yes/no.  I do appreciate that you have reached out to me in regards to
the issue, however, it turns out the issue was not with my code...my code
was correct the whole time.  The issue resided with the fact that the
website was just going through something at that time and would not accept
my code.   When I tried again with the same code, it was accepted.  This
was after 3 days of pulling my hair out THINKING it was my code.  Finding
this out was MORE frustratingm as it not only sent me on a wild goose
chase, but also added confusion to a learning student (myself) making me
think that my code was wrong.  Through this, I took a concept that I had
understanding about, thought it was then wrong, so led my train of
thought to think about the concept differently...taking me further from the
correct understanding and confusing me more.  I in fact totally lost the
concept that I had at first and luckily had the original code that I tried
to submit saved to my computer and was able to copy and paste it back in.
Later the moderator guided me to an area where they had updated in the
"frequently asked questions" area that stated that the vague message that I
was receiving (Error saving to server) was NOT due to poorly written code,
but that simply the server was having some sort of hiccup and couldn't save
at that time and to try again later.  I do not feel that this is adequate
to solving the problem that students are going to continue to run into.  I
mean I look at all the students complaining that there code isn't being
accepted and they think it is due to their coding.  This is not right.  If
this is the problem, there should be a different message displayed telling
the that it is not their code and to try back later (right there in the
autograder, etc.)  I mean I never would have found the amendment that the
moderator pointed me towards on my own and feel that a student should not
have to go on goose chase to find out that it is not their code.  Also the
chapter was vague and the videos did not cover enough to accomplish the
exercise (as admitted by the moderator).  Then this should be corrected to
adequately cover the material needed to accomplish the 4.6 assignment.
Rerecord that chapter with better information so not to just allow future
students to run into the same senario.

Thank you again for reaching out in attempt to rectify my issue.

Brian
